api-reference/10 UI Widgets/dxScheduler/1 Configuration/onAppointmentFormOpening.md

```diff
@@ -117,19 +161,134 @@ The following code shows how to use the **onAppointmentFormOpening** function to
+                if (!formItems.find(function(i) { return i.dataField === "location" })) {
+                    formItems.push({
+                        colSpan: 2,
+                        label: { text: "Location" },
+                        editorType: "dxTextBox",
+                        dataField: "location"
+                    });
+                    form.option("items", formItems);
+                }
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+    import Scheduler from 'devextreme-react/scheduler';
+
+    const dataSource = [{
+        text: "Attend the conference",
+        startDate: new Date(2020, 4, 24, 9, 10),
+        endDate: new Date(2020, 4, 24, 11, 20),
+    }];
 
-#include common-demobutton with {
-    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/Scheduler/CustomTemplates/"
-}
+    class App extends React.Component {
+        currentDate = new Date(2020, 4, 24);
+        
+        onAppointmentFormOpening(e) {
+            e.popup.option('showTitle', true);
+            e.popup.option('title', e.appointmentData.text ? 
+                e.appointmentData.text : 
+                'Create a new appointment');
+            
+            const form = e.form;
+            let mainGroupItems = form.itemOption('mainGroup').items;
+            if (!mainGroupItems.find(function(i) { return i.dataField === "phone" })) {
+                mainGroupItems.push({
+                    colSpan: 2, 
+                    label: { text: "Phone Number" },
+                    editorType: "dxTextBox",
+                    dataField: "phone"
+                });
+                form.itemOption('mainGroup', 'items', mainGroupItems);
+            }
+    
+            let formItems = form.option("items"); 
+            if (!formItems.find(function(i) { return i.dataField === "location" })) {
+                formItems.push({
+                    colSpan: 2,
+                    label: { text: "Location" },
+                    editorType: "dxTextBox",
+                    dataField: "location"
+                });
+                form.option("items", formItems);
+            }
+        }
+
+        render() {
+            return (
+                <Scheduler 
+                    dataSource={dataSource}
+                    defaultCurrentDate={this.currentDate}
+                    onAppointmentFormOpening={this.onAppointmentFormOpening}
+                />
+            );
+        }
+    }
+    export default App;
+
+---
 
+The `mainGroup` consists of two columns. To make a custom item span them both, set its [colSpan](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#colSpan) to 2 as done in the code above. Apply the same setting to an ungrouped item if it should span the `mainGroup` and `recurrenceGroup`. 
```

Comment:
```suggestion
The `mainGroup` consists of two columns. To make a custom item span both columns, set its [colSpan](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#colSpan) to '2' - as shown in the code above. Apply the same setting to an ungrouped item if it should span the `mainGroup` and `recurrenceGroup`. 
```

Reply:
Applied all these changes except for '2' - left it without quotes.

Reply:
Okay, thank you.